### PR TITLE
New hardcoded dataserver url points to a 308 Permanent Redirect

### PR DESCRIPTION
**Pull request by procule**
For some reason (which I guess is my old setup with a lot of customized things), the changed url implemented in 0cc07f7e5163870bcc2fb7281c28e8e39c9cbc54 was not working for me.

After looking at the code and using `gdb` I found out that the URL with the `ẁww.` prefix added is a `308 Permanent Redirect` to the url without the `www.` and it was resulting in an unsuccessful request. It may be related to the `libsoup` version I have installed (`2.64.2-2`) not following the redirect...

So long story short, removing the `www.` from the hardcoded URL did the trick and I think that pointing to the real URL would be best.

O.

### CODE REVIEW

**Comment by procule** [comment]
Also, according to [RFC7538, section 3](https://datatracker.ietf.org/doc/html/rfc7538#section-3), _The Hypertext Transfer Protocol Status Code 308 (Permanent Redirect)_:

> The 308 (Permanent Redirect) status code indicates that the target
   resource has been assigned a new permanent URI and **any future
   references to this resource ought to use one of the enclosed URIs**.
   Clients with link editing capabilities ought to automatically re-link
   references to the effective request URI ([Section 5.5 of [RFC7230]](https://datatracker.ietf.org/doc/html/rfc7230#section-5.5)) to
   one or more of the new references sent by the server, **where possible**.

**Comment by procule** [comment]
> It may be related to the `libsoup` version I have installed (`2.64.2-2`) not following the redirect...

Indeed, HTTP 308 Permanent Redirect was added in version `2.71.1` of `libsoup`:

https://gitlab.gnome.org/GNOME/libsoup/-/commit/51def076e937ce67f97f9a4db6765096b7661673#9f621eb5fd3bcb2fa5c7bd228c9b1ad42edc46c8

**Comment by procule** [comment]
@raveit65 can this PR be reviewed for merge ?

**Comment by procule** [comment]
GNOME's `libgweather` actually used the correct URL without the "www" in their patch: https://gitlab.gnome.org/GNOME/libgweather/-/commit/6c7aa4070b543f41e6f95978464dbef1ce467c41#2e1eaa99e66f946f2e3de5887adbc90fd2d263fc

And in their related issue, the changed url is also mentioned without "www": https://gitlab.gnome.org/GNOME/libgweather/-/issues/232

I guess the added "www" comes from that comment: https://github.com/mate-desktop/libmateweather/pull/79#issuecomment-1765810415

O.

**Review by raveit65 — APPROVED** [review]
Works fine with fedora 38 (libsoup-2.74.3-2.fc38.x86_64).

**Review by lukefromdc — APPROVED** [review]
A quick test with this installed gave weather information as expected

**Comment by procule** [comment]
> Works fine with fedora 38 (libsoup-2.74.3-2.fc38.x86_64).

A proper test would be with `libsoup < 2.71.1` (before their implementation of `308 Permanent Redirect` HTTP code)

**Comment by raveit65** [comment]
> > Works fine with fedora 38 (libsoup-2.74.3-2.fc38.x86_64).
> 
> A proper test would be with `libsoup < 2.71.1` (before their implementation of `308 Permanent Redirect` HTTP code)

I do not use a distro with such an old (obsolete) code base, and i assumed that you tested your PR with an old version, see first post.

New stable release
https://github.com/mate-desktop/libmateweather/releases/tag/v1.26.3
